src/main/java/lai/util/TaskList.java

```diff
@@ -0,0 +1,101 @@
+package lai.util;
+
+import lai.task.Task;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Represents a list of tasks in the Lai application.
+ * This class provides functionality to add, remove, and access tasks,
+ * as well as to iterate over all tasks.
+ */
+public class TaskList implements Iterable<Task> {
+    protected List<Task> tasks = new ArrayList<>();
+
+    /**
+     * Constructs a TaskList with a specified list of tasks.
+     *
+     * @param tasks The initial list of tasks.
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Adds a new task to the list and updates the storage file.
+     * Also prints a message indicating the task has been added.
+     *
+     * @param newTask The task to be added.
+     * @param storage The storage utility to update the tasks file.
+     * @return The updated list of tasks.
+     */
+    public List<Task> add(Task newTask, Storage storage) {
+        this.tasks.add(newTask);
+        storage.updateTasksFile(this);
+
+        Ui.printTaskAdded(newTask, this);
+
+        return this.tasks;
```

Comment:
Be careful, the caller may accidentally modify the internal variable `tasks` if you do this.